src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -158,6 +161,105 @@ static Status deserializeTensor(const std::string& requestedName, const KFSReque
+        // }
+    }
+    return mediapipe::ImageFormat::UNKNOWN;
+}
+
+static Status deserializeTensor(const std::string& requestedName, const KFSRequest& request, mediapipe::ImageFrame& outTensor) {
+    auto requestInputItr = request.inputs().begin();
+    auto status = getRequestInput(requestInputItr, requestedName, request);
+    if (!status.ok()) {
+        SPDLOG_DEBUG("Getting Input failed");
+        return status;
+    }
+    if (request.raw_input_contents().size() == 0 || request.raw_input_contents().size() != request.inputs().size()) {
+        std::stringstream ss;
+        ss << "Cannot find data in raw_input_content for input with name: " << requestedName;
+        const std::string details = ss.str();
+        SPDLOG_DEBUG("[servable name: {} version: {}] Invalid message structure - {}", request.model_name(), request.model_version(), details);
+        return Status(StatusCode::INVALID_MESSAGE_STRUCTURE, details);
+    }
+    auto inputIndex = requestInputItr - request.inputs().begin();
+    auto& bufferLocation = request.raw_input_contents().at(inputIndex);
+
+    if (requestInputItr->shape().size() != 3) {
+        SPDLOG_DEBUG("Invalid Mediapipe Image input shape size. Expected: 3 Actual: {}", requestInputItr->shape().size());
+        return Status(StatusCode::INVALID_SHAPE, "Invalid Mediapipe Image input shape size.");
+    }
+    size_t numberOfChannels = requestInputItr->shape()[2];
+    if (numberOfChannels == 0) {
+        SPDLOG_DEBUG("Invalid Mediapipe Image input number of channels. Number of channels should be greater than 0.");
+        return Status(StatusCode::INVALID_SHAPE, "Invalid Mediapipe Image input number of channels.");
+    }
+    size_t numberOfCols = requestInputItr->shape()[0];
+    size_t numberOfRows = requestInputItr->shape()[1];
+    size_t elementSize = KFSDataTypeSize(requestInputItr->datatype());
+    size_t expectedSize = numberOfChannels * numberOfCols * numberOfRows * elementSize;
+    if (bufferLocation.size() != expectedSize) {
+        SPDLOG_DEBUG("Invalid Mediapipe Image input buffer size. Actual: {} Expected: {}", bufferLocation.size(), expectedSize);
+        return Status(StatusCode::INVALID_CONTENT_SIZE, "Invalid Mediapipe Image input buffer size.");
+    }
+    auto imageFormat = KFSDatatypeToImageFormat(requestInputItr->datatype(), numberOfChannels);
+    if(imageFormat == mediapipe::ImageFormat::UNKNOWN){
+        SPDLOG_DEBUG("Invalid KFS request datatype, conversion to Mediapipe ImageFrame format failed.");
+        return Status(StatusCode::INVALID_INPUT_FORMAT, "Invalid KFS request datatype, conversion to Mediapipe ImageFrame format failed.");
+    }
+    size_t alignment = 1; // TODO check impact of used alignment on performance
+    auto outTensorFrame = mediapipe::ImageFrame(imageFormat, numberOfCols, numberOfRows, alignment);
+    std::memcpy(const_cast<uint8_t*>(outTensorFrame.PixelData()), bufferLocation.data(), expectedSize);
```

Comment:
Can we convert without a copy? We have guarantee that `bufferLocation` exists for entire run of the graph